src/openvic-simulation/dataloader/NodeTools.hpp

```diff
@@ -197,23 +223,39 @@ namespace OpenVic {
 			};
 		}
 
-		template<typename T, typename U>
-		callback_t<T> assign_variable_callback_cast(U& var) {
+		template<typename T>
+		Callback<T> auto assign_variable_callback_cast(auto& var) {
+			return [&var](T val) -> bool {
+				var = val;
+				return true;
+			};
+		}
+
+		template<std::integral T>
+		callback_t<T> assign_variable_callback_cast(auto& var) {
+			return [&var](T val) -> bool {
+				var = val;
+				return true;
+			};
+		}
+
+		template<std::signed_integral T>
+		callback_t<T> assign_variable_callback_cast(auto& var) {
```

Comment:
What are these specialisations for?

Reply:
Because of how these work, without this expec_int and expect_uint can't deduce the type you pass in, unfortunately since bool is an integral, this is actually a bit greedy.